src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoDataConnectAdapter.java

```diff
@@ -285,9 +286,8 @@ public TableData search(
                 "query", Optional.ofNullable(rewrittenQuery).orElse("(undefined)"))
         );
         JsonNode response = client.query(rewrittenQuery, extraCredentials);
-        QueryJob queryJob = createQueryJob(query, dataModel);
-        TableData tableData = toTableData(NEXT_PAGE_SEARCH_TEMPLATE, response, queryJob.getId(), request);
-        return tableData;
+        QueryJob queryJob = createQueryJob(response.get("id").asText(), query, response.get("nextUri").asText(), dataModel);
```

Comment:
@jvlasblom Any objection on using trino query id vs generating one ourself? Wanted to get your opinion as you are the original author of `query_job` table.

Reply:
No objection